src/api/app/controllers/webui/monitor_controller.rb

```diff
@@ -44,7 +44,8 @@ def update_building
   def events
     data = {}
 
-    arch = Architecture.find_by(name: params.fetch(:arch, @default_architecture))
+    arch = Architecture.find_by(name: params.fetch(:arch, @default_architecture).to_s)
```

Comment:
```suggestion
    arch = Architecture.find_by(name: params.fetch(:arch, @default_architecture))
```
Later on in this file we use `arch.name`...

Reply:
Opss... now I get it. `.to_s` is not applied to the result of the `find_by`. Sorry for the noise.